LoopKit/WeakObserverSet.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  WeakObserverSet.swift
+//  LoopKit
+//
+//  Created by Michael Pangburn
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+/// A set-like collection of weak types, providing closure-baesd iteration on a client-specified queue
+/// Mutations and iterations are thread-safe
+public class WeakObserverSet<Observer> {
```

Comment:
From the name alone, it isn't obvious what the difference between WeakObserverSet and WeakSet is. Something like `WeakSynchronizedSet<Element>` might be clearer; the observer pattern is the most common client of this class, but the functionality itself isn't inherently tied to use of observers.